libbeat/idxmgmt/ilm/ilm.go

```diff
@@ -45,6 +45,7 @@ type Supporter interface {
 // Manager uses a ClientHandler to install a policy.
 type Manager interface {
 	Enabled() (bool, error)
+	Overwrite() bool
```

Comment:
It's somewhat unfortunate that we have to extend the interface.

Better have the `Overwrite() bool` method on `Supporter`. The `Supporter` acts as a typed configuration object (e.g. it already exposes `Mode`, `Alias`, `Policy` settings) and as an abstract factory (`Manager` method) to create an instance based on the configuration object.

Maybe the name `Enabled` is somewhat missleading here. `Enabled` is not exposing a configuration setting, but is supposed to check if ILM support is available in the stack. We should renamed `Enabled` to `CheckEnabled` (proposal for better name is welcome) so to make it clear that this method is doing an active check against Elasticsearch (we can do it in a separate PR).